tests/unit/lib/transport/test_mqtt.py

```diff
@@ -0,0 +1,821 @@
+
+        # Encapsulate QoS related parameters
+        qos_details = QoSDetails(self.inflight, self.queue_size, self.retry)
+
+        mqtt_client = Mqtt(self.url, self.port, credentials, tls_conf, qos_details, self.client_id,
+                           self.client_clean_session, self.user_data, self.protocol, self.transport,
+                           self.keep_alive, self.enable_authentication, self.connection_disconnect_timeout)
+
+        # Publishing the message
+        mqtt_client.publish("test/publish", "Testing publish", 1, False)
+
+        # Check underline publish is called with correct arguments
+        mocked_publish.assert_called_with("test/publish", "Testing publish", 1, False)
+
+    @mock.patch.object(Mqtt, 'connect_soc')
+    @mock.patch.object(Client, 'subscribe')
+    @mock.patch.object(Client, "message_callback_add")
+    def test_subscribe(self, mocked_callback_add, mocked_subscribe, connect_soc):
+        """
+        Test case to test the subscribe method.
+        :return: None
+        """
+
+        # Encapsulate the authentication details
+        credentials = Identity(self.root_ca_cert, self.mqtt_username, self.mqtt_password,
+                               self.client_cert_file, self.client_key_file)
+
+        # Encapsulate TLS parameters
+        tls_conf = TLSConf(self.cert_required, self.tls_version, self.cipher)
+
+        # Encapsulate QoS related parameters
+        qos_details = QoSDetails(self.inflight, self.queue_size, self.retry)
+
+        mqtt_client = Mqtt(self.url, self.port, credentials, tls_conf, qos_details, self.client_id,
+                           self.client_clean_session, self.user_data, self.protocol, self.transport,
+                           self.keep_alive, self.enable_authentication, self.connection_disconnect_timeout)
+
+        # Subscribing to the topic
+        mqtt_client.subscribe("test/subscribe", 1, topic_subscribe_callback)
+
+        # Check underline subscribe is called with correct parameters
+        mocked_subscribe.assert_called_with("test/subscribe", 1)
+
+        # Check underline message_callback_add is called with correct parameters
+        mocked_callback_add.assert_called_with("test/subscribe", topic_subscribe_callback)
+
+    def test_QoS_class_implementation(self):
+        """
+        Test case to test the implementation of QoSDetails class.
+        :return: None
+        """
+
+        self.assertIsInstance(QoSDetails(self.inflight, self.queue_size, self.retry), QoSDetails,
+                              "Invalid implementation for QoSDetails class")
+
+    def test_mqtt_messaging_attributes_implementation_for_gateway_name(self):
+        """
+        Test case to test the implementation of MqttMessagingAttributes class for automatic topic generation.
+        :return: None
+        """
+        self.assertIsInstance(MqttMessagingAttributes(self.edge_system.name), MqttMessagingAttributes,
```

Comment:
This is not automatic topic generation test case.
Test case should be written similar to format data.  i.e., autogenerated pub and sub topics must be tested